platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/status/DefaultStatusStateMachine.java

```diff
@@ -179,19 +162,35 @@ public void processStatusAction(@NonNull final PlatformStatusAction action) {
                 () -> new PlatformStatusPayload(statusChangeMessage, previousStatusName, newStatusName).toString());
 
         currentStatusLogic = newLogic;
-        currentStatus.set(currentStatusLogic.getStatus());
 
+        final PlatformStatus newStatus = currentStatusLogic.getStatus();
         currentStatusStartTime = time.now();
 
-        statusChangeConsumer.accept(newLogic.getStatus());
+        return newStatus;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Nullable
+    @Override
+    public PlatformStatus issOccurred(@NonNull final IssNotification issNotification) {
+        if (Set.of(IssNotification.IssType.SELF_ISS, IssNotification.IssType.CATASTROPHIC_ISS)
+                .contains(issNotification.getIssType())) {
+
+            return submitStatusAction(new CatastrophicFailureAction());
+        }
+
+        // don't change status for other types of ISS
+        return null;
     }
 
     /**
      * {@inheritDoc}
      */
+    @Nullable
     @Override
-    @NonNull
-    public PlatformStatus getCurrentStatus() {
-        return currentStatus.get();
+    public PlatformStatus timeElapsed(@NonNull final Instant time) {
+        return submitStatusAction(new TimeElapsedAction(time));
     }
```

Comment:
I wonder if it would make sense to rename this to `heartbeat()` so that it's more obvious the way this is used. (Optional suggestion, I defer to your final decision on naming here.)

Reply:
done